### DISCUSSION TITLE
Minimal Pystac STAC

### DISCUSSION THREAD

**Discussion by ljstrnadiii** [gjoseph92/stackstac#110] [Q&A] [answered]
Another question as I try to get started with stackstac. 

I am hoping to build a small collection/catalog for a set of 20ish zoom 10 tiff files that are contiguous in space. Is there any goal or example of building a minimal STAC catalog for something like this to demonstrate usage with stackstac?

I am not exactly sure what are the minimal set of attributed for each item in order to operate with stackstac correctly. Any guidance would be greatly appreciated!

**Comment by gjoseph92** [comment, marked as answer]
You're trying to make a standalone/static STAC catalog? I think @scottyhq has been playing with this, see https://github.com/gjoseph92/stackstac/discussions/86.

Generally the idea of a single zipfile you can download that includes both a STAC and some assets could be helpful (also would be helpful for stackstac's tests). I've wanted this at times. Currently I think people have more commonly been using the free public datasets hosted on AWS or [Azure](https://planetarycomputer.microsoft.com/) to try out stackstac.

If you just want to experiment with things, there's also a link to [try stackstac in binder](https://mybinder.org/v2/gh/gjoseph92/stackstac/main?labpath=%2Fdocs%2Fbasic.ipynb%3Ffile-browser-path%3D%2Fexamples) on the README. Or I highly recommend signing up for Planetary Computer and using the Hub, which includes some nice example notebooks, has a prebuilt software environment, and lets you spin up dask clusters easily.

> I am not exactly sure what are the minimal set of attributed for each item in order to operate with stackstac correctly. Any guidance would be greatly appreciated!

I should make a reference page for this on the docs. For best results, you'll want your STAC items to:
1. Use the [`proj`](https://github.com/stac-extensions/projection) extension, specifically:
    1. `proj:epsg`
    2. `proj:transform`
    3. `proj:bbox`
    3. `proj:shape`
1. Have single-band assets
2. Specify the `type` of each asset (`image/tiff`, etc.)

**Reply by ljstrnadiii** [reply]
Sweet. Thanks for sharing. 

Yeah, I guess I am trying to build a static STAC catalog. This is all very new to me, so I am just getting familiar to be honest. We get out tiff files from earth engine and have been talking about building a stac catalog for regions we download data for. I suppose it would be a static catalog on gcs and update when we add a new collection if that ends up making sense. 

Thanks for the details! I'll make sure to add.

**Reply by gjoseph92** [reply]
You could look at https://github.com/developmentseed/rio-stac; it's a utility for generating STAC metadata out of GeoTIFFs. Only works on a single file, but combined with `pystac` you could probably make something to generate metadata for a whole catalog of files. I imagine there are tools/examples of doing this, but I'm not familiar with them.
